xdsl/dialects/riscv.py

```diff
@@ -2491,6 +2491,7 @@ class LiOp(RISCVInstruction, ABC):
 
     rd = result_def(IntRegisterType)
     immediate = attr_def(base(Imm32Attr) | base(LabelAttr))
+    fastmath = opt_attr_def(FastMathFlagsAttr)
```

Comment:
What is the semantics change of li being optionally fastmath?

Reply:
I spent a while trying to find any information on this, and I'm not seeing that there is any actual effect on the `li` instr at all when compiled with fastmath. `Chapter 15. Load Immediate` of the [riscv-asm-manual](https://github.com/riscv-non-isa/riscv-asm-manual.git) doesn't mention fastmath at all, and [godbolt](https://godbolt.org/z/Yf4jP4z8d) doesn't seem to show any difference when compiled with/without `-ffast-math`. 

Is there anywhere else that I should look for information on this? If not, should I go ahead and remove the `fastmath` attribute from `LiOp`?

Reply:
I also don't know how it would affect anything. I don't think arith.constant can have fastmath, can it? It seems safe to remove from this PR, and add it later if we find a way to leverage it.

Reply:
I've removed it now, and rebased everything into one commit since it was getting messy